src/stm32/gpio.h

```diff
@@ -55,4 +55,23 @@ void i2c_write(struct i2c_config config, uint8_t write_len, uint8_t *write);
 void i2c_read(struct i2c_config config, uint8_t reg_len, uint8_t *reg
               , uint8_t read_len, uint8_t *read);
 
+struct sdio_config {
+    void *sdio;
+};
+struct sdio_config sdio_setup(uint32_t bus);
+void sdio_send_cmd(struct sdio_config sdio, uint8_t cmd, uint32_t argument
+                   , uint8_t wait);
+uint8_t sdio_send_command(struct sdio_config sdio_config, uint8_t cmd
+                          , uint32_t argument, uint8_t wait
+                          , uint8_t *response_data
+                          , uint8_t *response_data_len);
+uint8_t sdio_prepare_data_transfer(struct sdio_config sdio_config, uint8_t read
+                                   , uint32_t numblocks, uint32_t blocksize);
+uint8_t sdio_read_data(struct sdio_config sdio_config, uint8_t *data
+                       , uint32_t numblocks, uint32_t blocksize);
+uint8_t sdio_write_data(struct sdio_config sdio_config, uint8_t *data
+                        , uint32_t numblocks, uint32_t blocksize);
+void sdio_switch_bus_width_and_speed(struct sdio_config sdio_config
+                                     , uint8_t width, uint8_t clkdiv);
+
```

Comment:
I'd add these to a new "sdio.h" file.  I don't think we should modify "gpio.h" for this.

Reply:
Proposed changes included in latest commit.